core/src/Cuda/Kokkos_Cuda_Half.hpp

```diff
@@ -943,6 +1092,25 @@ KOKKOS_INLINE_FUNCTION
 }
 #endif
 }  // namespace Experimental
+
+// use float as the return type for sum and prod since cuda_fp16.h
+// has no constexpr functions for casting to __half
```

Comment:
This comment raises an issue that should be addressed in the compiler and standard library - I'll call Max Katz's attention to it.

Reply:
Acknowledged and requested for cuda_fp16.h.